### PR TITLE
feat(server): add new server module

### PR DESCRIPTION
**Pull request by fieldju**

### CODE REVIEW

File: iam/README.md [hunk 1]
```diff
@@ -37,7 +37,7 @@ Pass in optional validators for verifying specific scopes or other properties on
 
 ```go
 // myCustomValidator checks if the user can access deployment information.
-func myCustomValidator(p *ArmoryCloudPrincipal) error {
+func myCustomValidator(p *ArmoryCloudPrincipal) bs {
```
Comment (danielpeach) [inline iam/README.md:40]: I don't think `bs` is right?

File: server/error.go [hunk 2]
```diff
@@ -0,0 +1,378 @@
...
+	Errors  []ResponseContractErrorDTO `json:"errors"`
+}
+
+type ResponseContractErrorDTO struct {
+	Message  string         `json:"message"`
+	Metadata map[string]any `json:"metadata,omitempty"`
+	Code     string         `json:"code"`
+}
+
+// APIError an error that gets embedded in ResponseContract when an error response is return to the client
```
Comment (danielpeach) [inline server/error.go:38]:
```suggestion
// APIError is an error that gets embedded in ResponseContract when an error response is returned to the client
```

File: iam/README.md (hunk 1 above)
Comment (fieldju) [reply iam/README.md:40]: yeah IDEAs refactor with golang sometimes does some whacky stuff.

File: server/error.go [hunk 3]
```diff
@@ -0,0 +1,378 @@
...
+const defaultErrorCode = 42
+
+var sensitiveHeaderNamesInLowerCase = []string{
+	"authorization",
+	"x-armory-proxied-authorization",
+}
+
+// ResponseContract the strongly typed error contract that will be returned to the client if a request is not successful
+type ResponseContract struct {
+	ErrorId string                     `json:"error_id"`
```
[the remark this answers is not in the document]
Comment (fieldju) [reply server/error.go]: image

But with our goal of deprecating Java, and the fact that its pretty much a feature complete lib that I have already forked to fix a couple things, there isn't a reason I can't make it camel case or write a customer jackson serializer and update our projects to use it.

Do you want me to make this all camel case and we roll forward with that and I'll patch the java services?

**Pull request merged**